python-package/lightgbm/compat.py

```diff
@@ -146,6 +149,18 @@ class dask_Array:  # type: ignore
 
         pass
 
+    def dask_array_concatenate(*args, **kwargs):  # type: ignore
+        """Mock ``dask.array.concatenate()``."""
```

Comment:
note for reviewers: I used "Mock" instead of being consistent with the use of "Dummy" elsewhere in this file because `pycodestyle` is not happy with "Dummy" for function docstrings.

Failed lint job: https://github.com/microsoft/LightGBM/pull/4378/checks?check_run_id=2836021678

> ./python-package/lightgbm/compat.py:153 in public function `dask_array_concatenate`:
        D401: First line should be in imperative mood; try rephrasing (found 'Dummy')
./python-package/lightgbm/compat.py:157 in public function `dask_array_from_delayed`:
        D401: First line should be in imperative mood; try rephrasing (found 'Dummy')
./python-package/lightgbm/compat.py:161 in public function `dask_bag_from_delayed`:
        D401: First line should be in imperative mood; try rephrasing (found 'Dummy')